src/components/cron.ts

```diff
@@ -126,3 +129,28 @@ export const createCoffeeChatCron = (client: Client): CronJob =>
       throw 'Bad channel type';
     }
   });
+
+export const assignCodeyRoleForLeaderboard = (client: Client): CronJob =>
+  new CronJob('0 0 * * * *', async function () {
+    const leaderboard = await getCoinLeaderboard(10);
+    const guild = client.guilds.resolve(TARGET_GUILD_ID);
+    if (!guild) {
+      throw 'guild not found';
+    }
+    const members = await guild.members.fetch();
+    // Removing role from previous users
+    const usersPreviousRole = await loadRoleUsers('1057399983357313175'); // use role id, not the actual name
+    const guildMembersPreviousRole = usersPreviousRole.map((user) => members.get(user.id));
+    guildMembersPreviousRole.forEach(async (user) => {
+      if (user != undefined) {
+        await updateMemberRole(user, 'codeyCoin', false);
```

Comment:
@mcpenguin Hey I am not sure what you mean by getting the name of the role here from the config id? I thought that since the name of the role was supposed be CodeyCoin, I thought we could hard code it. Not sure how I would do that.

Reply:
It might not be "CodeyCoin", we might change it to "CodeyCoin Top 10" or something. We should always make sure that we avoid hard coded values, since it makes our code inflexible.